### PR TITLE
feat: ESM migration of got-scraping

### LINKED ISSUE #90: Type definitions for "gotScraping()" are wrong/missing methods:
```
  1 import { gotScraping } from "got-scraping";
  2
  3 async function main() {
  4   let x = gotScraping("https://www.example.com");
  5
 >>   console.log(x.then);  Property 'then' does not exist on type 'CancelableRequest<Response<string>>'.
  7 }
  8
  9 main();
```

### LINKED ISSUE #105: Invalid charset error in url
Urls with invalid charset such as 

https://datadeliver.net/ads.txt has set `uft-8` instead of `utf-8` results in error

```
ERROR HttpCrawler: Request failed and reached maximum retries. Error: Resource https://datadeliver.net/ads.txt served with unsupported charset/encoding: uft-8
    at HttpCrawler._encodeResponse (/node_modules/@crawlee/http/internals/http-crawler.js:544:15)
    at HttpCrawler._parseResponse (/node_modules/@crawlee/http/internals/http-crawler.js:442:45)
    at HttpCrawler._runRequestHandler (/node_modules/@crawlee/http/internals/http-crawler.js:308:39)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async wrap (/node_modules/@apify/timeout/index.js:52:21) {"id":"uIK9UTUGCjKrRhO","url":"https://datadeliver.net/ads.txt","method":"GET","uniqueKey":"https://datadeliver.net/ads.txt"}
```

It works fine on curl and browser

It is expected to have such error or we can skip such checks?

### LINKED ISSUE #106: Using gotScraping.extend results in incorrect type information
```ts
import { gotScraping } from 'got-scraping';

const instance = gotScraping.extend();

const response = await instance.get('https://api.apify.com/v2/browser-info', {
    responseType: 'json',
    proxyUrl: 'proxy'
});
console.log(response.body);
```

I had no doubt that I could write something like this, but when I actually run this code in TypeScript, it prints an error due to type information.

I ran the same code in Javascript, but the only thing that is wrong is the type information because Proxy worked properly

### LINKED ISSUE #108: ESM rewrite
## Why

`got` is ESM only, and that itself imposes issues on using the latest versions for us, especially on its dependencies.

## What

- rewrite the package to be ESM only
- update to latest `got`
- use the version in `crawlee` (and potentially other places) via dynamic imports

### PR DESCRIPTION
**Pull request by vladfrangu**
Closes #90 
Closes #106 
Closes #108 
Closes #105 

BREAKING CHANGE
This package is now ESM only, as such, it cannot be required from inside a CJS project, use dynamic import instead.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,52 +2,58 @@
...
+    "type": "module",
     "dependencies": {
-        "got-cjs": "12.5.4",
-        "header-generator": "^2.1.3",
-        "http2-wrapper": "^2.1.11",
-        "mimic-response": "^3.1.0",
-        "ow": "^0.28.1",
-        "quick-lru": "^5.1.1",
-        "tslib": "^2.4.0"
+        "got": "13.0.0",
```
Comment (B4nan) [inline package.json:19]: i guess we dont need to pin the dependency here

File: .eslintrc.json [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
+	"overrides": [
+		{
+			"files": ["./test/**/*.ts"],
+			"rules": {
+				"@typescript-eslint/no-explicit-any": "off"
+			}
+		}
+	],
+	"rules": {
+		"@typescript-eslint/no-explicit-any": "warn"
```
Comment (B4nan) [inline .eslintrc.json:18]: we can remove this for the sources as well, i'd rather not have any warnings in the log. or you can resolve the warnings instead, will leave it up to you

File: src/resolve-protocol.ts [hunk 3]
```diff
@@ -12,7 +12,8 @@ const connect = async (proxyUrl: string, options: tls.ConnectionOptions, callbac
         host = `[${options.host}]:${options.port}`;
     }
 
-    (async () => {
+    // eslint-disable-next-line no-void
```
Comment (B4nan) [inline src/resolve-protocol.ts:15]: we really need to get rid of the `no-void` rule as well

Comment (vladfrangu) [reply src/resolve-protocol.ts:15]: I've mentioned https://github.com/iCrawl/eslint-config-neon before, our eslint rules really need some love and attention 😢

File: package.json (hunk 1 above)
Comment (vladfrangu) [reply package.json:19]: Good catch, we can use `^` for the version range

File: tsconfig.json [hunk 4]
```diff
@@ -1,7 +1,11 @@
 {
 	"extends": "@apify/tsconfig",
 	"compilerOptions": {
-		"outDir": "dist"
+		"outDir": "dist",
```
Comment (B4nan) [inline tsconfig.json:4]: we should also add `"target": "ES2022"` here

Comment (vladfrangu) [reply tsconfig.json:4]: Can't wait for when we can bump it at the /tsconfig level \o/

**Review by B4nan — APPROVED** [review]

**Pull request merged**